sdk/python/feast/feature_server.py

```diff
@@ -88,10 +90,15 @@ def get_online_features(body=Depends(get_body)):
             body = json.loads(body)
             # Initialize parameters for FeatureStore.get_online_features(...) call
             if "feature_service" in body:
-                features = store.get_feature_service(
+                feature_service = store.get_feature_service(
                     body["feature_service"], allow_cache=True
                 )
+                assert_permissions(
+                    resources=feature_service, actions=[AuthzedAction.QUERY_ONLINE]
+                )
+                features = feature_service
             else:
+                # TODO: How add the permissions if it has feature list ??
```

Comment:
Quoting @tokoko:
>  I think in practice it would make sense to try to fully decouple that process from the execution even if it means that some of the objects will need to be fetched twice from the registry (once during permission check by server code and again during execution by provider code).

In short, we need to look at the implementationin FeatureStore and anticipate the fetch of related (and secured) resources

Reply:
Should be just these 2 steps, I think. Suppose user requested `features=["driver_hourly_stats:conv_rate", "driver_hourly_stats:acc_rate", "driver_daily_features:daily_miles_driven"]`

1. Because we are tracking permissions on feature view-level, not feature-level, we need to extract a list of feature views requested from above. It will be `["driver_hourly_stats", "driver_daily_features"]` in this instance.
2. We need to get each FV from the registry and check `QUERY_ONLINE` permission on each (`QUERY_OFFLINE` in flight server)

Reply:
@tokoko any idea if there any util function which fetch fv based on features or similar

Reply:
@redhatHameed [this](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/utils.py#L763C5-L763C30) is the closest thing, although it's returning two lists instead of one. btw, the implementation will probably change after #4328 is merged.

Reply:
@tokoko Thanks, I utilize same function update the code, by ignoring other list, let me know WDYT ? if there is need to create a separated function or we can continue with that. 

```
all_feature_views, _ = utils._get_feature_views_to_use(
                    store.registry,
                    store.project,
                    features,
                    allow_cache=True,
                    hide_dummy_entity=False,
                )
                for feature_view in all_feature_views:
                    assert_permissions(
                        resource=feature_view, actions=[AuthzedAction.QUERY_ONLINE]
                    )
```

Reply:
I don't think you should ignore the second one, you should assert permissions on both.

Reply:
updated.